config/eslint/eslintrc.js

```diff
@@ -0,0 +1,205 @@
+          },
+          String: {
+            message: "Avoid using the `String` type. Did you mean `string`?",
+          },
+          Symbol: {
+            message: "Avoid using the `Symbol` type. Did you mean `symbol`?",
+          },
+        },
+        extendDefaults: false,
+      },
+    ],
+    "@typescript-eslint/consistent-type-assertions": "error",
+    "@typescript-eslint/consistent-type-definitions": "error",
+    "@typescript-eslint/dot-notation": "error",
+    "@typescript-eslint/explicit-member-accessibility": [
+      "error",
+      {
+        accessibility: "explicit",
+        overrides: {
+          constructors: "no-public",
+        },
+      },
+    ],
+    "@typescript-eslint/naming-convention": [
+      "error",
+      {
+        selector: "default",
+        format: ["camelCase"],
+        leadingUnderscore: "allow",
+        trailingUnderscore: "allow",
+      },
+      {
+        selector: ["variable", "parameter"],
+        format: ["camelCase", "UPPER_CASE", "PascalCase"],
+        leadingUnderscore: "allow",
+        trailingUnderscore: "allow",
+      },
+      {
+        selector: "classProperty",
+        format: ["camelCase", "UPPER_CASE"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "enumMember",
+        format: ["UPPER_CASE"],
+      },
+      {
+        selector: "memberLike",
+        modifiers: ["private"],
+        format: ["camelCase"],
+        leadingUnderscore: "require",
+      },
+      {
+        selector: ["objectLiteralProperty", "objectLiteralMethod"],
+        format: ["camelCase", "PascalCase", "snake_case", "UPPER_CASE"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "typeProperty",
+        format: ["camelCase", "PascalCase"],
+        leadingUnderscore: "allow",
+      },
+      {
+        selector: "typeLike",
+        format: ["PascalCase"],
+      },
+      {
+        selector: "typeProperty",
+        filter: "__hardhatContext",
+        format: null,
+      },
+    ],
+    "@typescript-eslint/no-empty-interface": "error",
+    "@typescript-eslint/no-floating-promises": "error",
+    "@typescript-eslint/no-misused-new": "error",
+    "@typescript-eslint/no-namespace": "error",
+    "@typescript-eslint/no-redeclare": "error",
+    "@typescript-eslint/no-shadow": [
+      "error",
+      {
+        hoist: "all",
+      },
+    ],
+    "@typescript-eslint/no-this-alias": "error",
+    "@typescript-eslint/no-unused-expressions": "error",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      {
+        argsIgnorePattern: "^_",
+        varsIgnorePattern: "^_",
+      },
+    ],
+    "@typescript-eslint/prefer-for-of": "error",
+    "@typescript-eslint/prefer-function-type": "error",
+    "@typescript-eslint/prefer-namespace-keyword": "error",
+    "@typescript-eslint/restrict-plus-operands": "error",
+    "@typescript-eslint/strict-boolean-expressions": [
+      "error",
+      {
+        allowAny: true,
```

Comment:
I think this should be set to `false`, but it'd require a few changes. Should we open an issue about this?

Reply:
Agree :+1: https://github.com/nomiclabs/hardhat/issues/1595